Script to build and configure nfp along with the required

hemanthravi opened the pull request:
disk image builder scripts.

@njagadish @mabaig043 @rajendramachani @mak-454 @songole @rayalaman 

The following needs to be done before submitting to openstack
1. Merge contents of gbpservice/tests/contrib/diskimage-create into gbservice/contrib/nfp/tools/image_builder
2. There should be only 1 disk_image_create.py
3. Delete configurator_conf.json. This appears to be unused.
4. Modify gbpservice/tests/contrib/nfp for the new location of disk_image_create
5. Any other users of disk_image_create?
1. templates/*.template should be converted to yaml

hemanthravi committed:
Script to build and configure nfp along with the required
disk image builder scripts.

rajendramachani commented on gbpservice/contrib/nfp/tools/image_builder/Dockerfile:
```diff
@@ -0,0 +1,64 @@
+ git\
+ rabbitmq-server\
+ python-dev \
+ screen
```
Don't we need 'vim' and 'unzip' package here ?

They are there in gbpservice/contrib/nfp/configurator/Dockerfile

aijaz-baig committed:
Review Comment Fixes

>> Removed docker load from configurator user-data
>> Detecting the operating system type (ubuntu/centos) and selecting
   proper path to get src_dirs
>> Added cache dir support for image builder

aijaz-baig committed:
Added executable permission for elements

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/Dockerfile:
```diff
@@ -0,0 +1,64 @@
+COPY ./neutron_lbaas /usr/local/lib/python2.7/dist-packages/neutron_lbaas
+COPY ./configurator_run.sh /configurator_run.sh
+
+# removing loadbalance v2 agent, driver, and paloalto fw driver
```
There will not be enterprise code in opensource installation. No need to remove these files. Remove following 3 lines.

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/disk_image_create.py:
```diff
@@ -0,0 +1,138 @@
+    # root login enabled, set password environment varaible
+    if 'root-passwd' in dib['elements']:
+        os.environ['DIB_PASSWORD'] = dib['root_pswd']
+    # element for creating visibility image
```
Correct this comments. Remove visibility keyword.

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/elements/configurator/README.md:
```diff
@@ -0,0 +1,5 @@
+Visibility
```
Remove this visibility keyword.

njagadish commented on gbpservice/contrib/nfp/tools/setup_nfp.py:
```diff
@@ -0,0 +1,498 @@
+def launch_configurator():
+    get_openstack_creds()
+    if os.path.isfile(args.controller_path):
+        os.system("glance image-create --name configurator --disk-format qcow2  --container-format bare  --visibility public --file " + args.controller_path )
```
What is this --visibility argument?

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/disk_image_create.py:
```diff
@@ -0,0 +1,138 @@
+import subprocess
+import sys
+
+conf = {}
```
Lets us have a class in this file to contain all the methods and maintain the conf and any other variables as class member variables. Lets do this as part of fixing devstack.

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/elements/root-passwd/README.rst:
```diff
@@ -0,0 +1,5 @@
+This element assigns a password to the root account in the image and enables
+password login via ssh.
```
Lets remove the root-passwd directory as there is password-less login with ssh private/public key for ubuntu user.

njagadish commented on gbpservice/contrib/nfp/tools/image_builder/disk_image_create.py:
```diff
@@ -0,0 +1,138 @@
+    dib_args = ['disk-image-create', 'base', 'vm', 'ubuntu']
+    image_name = conf['ubuntu_release']['release']
+    # root login enabled, set password environment varaible
+    if 'root-passwd' in dib['elements']:
```
We can remove the root-passwd element dependency.

rayalaman committed:
Fixed Review Comments

>> Removed enterprise references from Dockerfile
>> removed root-passwd from image builder

rajendramachani committed:
Fixed devstack to use nfp/tools/image_builder
Addressed few pep8 warning

rajendramachani committed:
Fixed configurator source preperation
Fixed CIDRs for ip pools

rajendramachani committed:
Fixed syntax error

rajendramachani committed:
Fixed git branch for configurator sources

rayalaman committed:
Dockerfile

>> removed unneeded 'git' package

rayalaman committed:
Dockerfile

>> removed unneeded 'git' package from Dockerfile

hemanthravi commented on devstack/lib/nfp:
```diff
@@ -196,8 +196,8 @@ function create_nfp_gbp_resources {
  --ip-version 4\
- --proxy-ip-pool=192.169.0.0/24\
- --ip-pool 120.0.0.0/24\
+ --proxy-ip-pool=192.168.0.0/16\
```
Remove --proxy-ip-pool here. It's ok to pick the default (defined in ./neutron/extensions/driver_proxy_group.py)

hemanthravi commented on devstack/lib/nfp:
```diff
@@ -244,7 +244,12 @@ function create_nfp_image {
-        sudo python $DISKIMAGE_CREATE_DIR/disk_image_create.py $DISKIMAGE_CREATE_DIR/ref_configurator_conf.json
+        sudo python -c\
+ 'from gbpservice.contrib.nfp.tools.image_builder import disk_image_create as DIB;\
+ DIb.cur_dir = $DISKIMAGE_CREATE_DIR\
```
Should this be DIB ?

hemanthravi commented on devstack/lib/nfp:
```diff
@@ -255,7 +260,31 @@ function create_nfp_image {
+            rm -rf neutron-lbaas
+            sudo python -c\
+ 'from gbpservice.contrib.nfp.tools.image_builder import disk_image_create as DIB;\
+ DIb.cur_dir = $DISKIMAGE_CREATE_DIR\
```
Should this be DIB?

hemanthravi commented on devstack/plugin.sh:
```diff
@@ -35,8 +35,8 @@ function nfp_configure_neutron {
-    iniset $NEUTRON_CONF group_policy_implicit_policy default_ip_pool "11.0.0.0/8"
-    iniset $NEUTRON_CONF group_policy_implicit_policy default_proxy_ip_pool "192.169.0.0/16"
+    iniset $NEUTRON_CONF group_policy_implicit_policy default_ip_pool "172.16.0.0/12"
+    iniset $NEUTRON_CONF group_policy_implicit_policy default_proxy_ip_pool "192.168.0.0/16"
```
OK to use the defaults defined in:
./neutron/services/grouppolicy/drivers/implicit_policy.py
./neutron/extensions/driver_proxy_group.py

The 2 lines are not required

hemanthravi committed:
Remove overrides that are not required.

hemanthravi committed:
Use 172.17.0.0/16 for service management l3p proxy-ip-pool instead
of default. Otherwise ptg creation will fail due to an overlap of
proxy-ip-pool for the implicitly created l3p.

rajendramachani committed:
Fixed disk image create script invocation

rajendramachani committed:
Fixed shell command substitution

rajendramachani committed:
FIxed command substitution

rajendramachani committed:
Adding __init__.py to fix python package

rajendramachani committed:
Fixed home directory for cache

rajendramachani committed:
Fixed neutron package source branch for configurator docker

rajendramachani committed:
Fixed gate test and addressed few pep8 warnings

rajendramachani committed:
Fixed devstack install, for cloning gbpservice from gerrit ref
Addressed review comments on gerrit changeset
Fixed the gate test

rajendramachani committed:
Fixed devstack install

aijaz-baig committed:
Docker Build

>> Moved configurator docker build to /tmp directory
>> Issue noticed if setup_nfp.py was invoked from
   /usr/lib/python2.7/site-packages/gbpservice/.... as a
   recursive copy of gbpservice/ into iteself fails

aijaz-baig committed:
Docker Build

>> setting the docker_build_dir incase the script is not invoked
   from setup_nfp.py

njagadish committed:
Merge pull request #588 from oneconvergence/nfp-config-scripts_docker

Docker Build
